metadata_upload/metadata_upload_tutorial.Rmd

```diff
@@ -0,0 +1,623 @@
+  stopifnot(
+    class(con) == "PqConnection",
+    is.character(query),
+    length(query) == 1,
+    is.list(params)
+  )
+  qry <- DBI::dbSendStatement(con, query)
+  res <- DBI::dbBind(qry, params)
+  on.exit(DBI::dbClearResult(res))
+}
+# get_fields
+# this function will be run on a row of a google sheet (experiments, sites, treatments, etc.) 
+# will return a vector containing names of user completed fields that are also a field in a bety table
+# this output will be needed to generate parameterized queries
+get_fields <- function(row, tbl_fields){
+  ind <- which(!is.na(row))
+  field <- names(row)[ind]
+  bety_field <- field[field %in% tbl_fields]
+  return(bety_field)
+}
+# get_statement
+# this function will generate a parameterized insert statement for a row
+get_statement <- function(row, fields, tbl_name){
+  position_params <- sapply(1:length(fields),
+                       function(x) paste0('$', x),
+                       USE.NAMES = FALSE)
+  statement <- paste0("insert into ",
+                  tbl_name,
+                  " (",
+                  paste(fields, collapse = ', '),
+                  ") values (",
+                  paste(position_params, collapse = ', '),
+                  ")")
+  return(statement)
+}
+# get_params
+# this function will return an unnamed list of parameter values for a row according to field names passed in
+get_params <- function(row, fields){
+  params <- unname(as.list(row[fields]))
+  return(params)
+}
+# fetch_query
+# will be used to fetch metadata from test database 
+# this function can be used to check success of metadata upload
+fetch_query <- function(con, query){
+  qry <- DBI::dbSendQuery(con, query)
+  on.exit(DBI::dbClearResult(qry))
+  DBI::dbFetch(qry)
+}
+```
+
+## Step 1: Add experiments
+
+Run the following chunk to add new experiments
+
+```{r add-new-exp}
+exp_fields <- c('name',
+                'start_date',
+                'end_date',
+                'description',
+                'design',
+                'user_id')
+# loop through each row of experiments 
+for(i in 1:nrow(experiments)){
+  row <- experiments %>% slice(i)
+  fields <- get_fields(row = row,
+                       tbl_fields = exp_fields)
+  statement <- get_statement(row = row,
+                             fields = fields,
+                             tbl_name = 'experiments')
+  params <- get_params(row = row,
+                       fields = fields)
+  prepared_statement(con = dbcon,
+                     query = statement,
+                     params = params)
+}
+```
+
+### Check if new experiments have been successfully uploaded
+
+Run this chunk to see if your new experiments have been successfully added to the test database
+
+```{r new-exp-check}
+exp_query <- glue::glue_sql("select * from experiments where name in 
+                            ({experiments$name*})",
+                            .con = dbcon)
+fetch_query(con = dbcon,
+            query = exp_query)
```

Comment:
* present as a table 
* also test that this is the same as the information that came from the spreadsheet; consider using a package like https://github.com/ropensci/assertr (or maybe @KristinaRiemer or @jeanetteclark have other thoughts on this?)